ArduCopter/mode.cpp

```diff
@@ -347,6 +347,20 @@ bool Copter::set_mode(Mode::Number mode, ModeReason reason)
         return false;
     }
 
+#if AP_FENCE_ENABLED
+    // may not be allowed to change mode if recovering from fence breach
```

Comment:
if you talk to regulators then you may find they are horrified by the copter behaviour. I recently had a call with one of the australian regulators and they asked about this

Reply:
I don't mind if people think there is a use case, but I would object if it were the default behavior as it is on plane.

Its the same argument as we quite often have. Setting this option assumes the operator does not know what they are doing. It removes all options except termination.

I'm sure you have seen cases were it is very bad too. The obvious one is a GPS glitch. Your flying along quite happy, you get a GPS glitch. Vehicle now thinks its outside the fence it triggers RTL and starts flying off to the incorrect location and prevents you from taking over in anyway that could recover the vehicle. You now have to terminate the vehicle.

Reply:
Yes, it's a bit of a golden rule (at least on copter and rover) and we don't second guess what the pilot is requesting

Reply:
> I'm sure you have seen cases were it is very bad too. The obvious one is a GPS glitch. Your flying along quite happy, you get a GPS glitch. Vehicle now thinks its outside the fence it triggers RTL and starts flying off to the incorrect location and prevents you from taking over in anyway that could recover the vehicle. You now have to terminate the vehicle.

This is why I always fly with fence-enable/disable on a switch.  Because the default behaviour without enable/disable on a switch is terrifying.

Reply:
The biggest problem with this on plane is that fence breach/RTL will quite likely fly further out of the fence as it tries to recover from the breach, I'm sure the regulators will be horrified when a fence breach causes the vehicle to fly more directly towards the building, airport or crowd, with no way for the pilot to correct it. I have lost several planes because of this. [But I'm waiting patiently for this PR to go in before suggesting improvements, so please don't hold it up].

Reply:
@IamPete1 the default is off on copter. The current fence behaviour is known to be frowned upon by the CAA, so I think adding it is reasonable.